### ISSUE TITLE
Support ZIM files with Service Worker

### ISSUE THREAD

**Issue by kelson42** [kiwix/apple#341]
New ZIMit based ZIM file use [Service Worker](https://developer.mozilla.org/en-US/docs/Web/API/Service_Worker_API).

**Comment by automactic** [comment]
TLDR: I don't think this is really possible right now.

In iOS 14, browser apps can become the default browser by using `com.apple.developer.web-browser` entitlement. (e.g. use Firefox as the default browser on iPhone) This AFAIK is the only way to have `WkWebView` support service workers right now. 

Kiwix is not a generic purpose internet browser and cannot use this entitlement (I can positively say the app will be rejected if we do it), therefore there doesn't seem to be a way to support this request

See apple doc for more: [link](https://developer.apple.com/documentation/xcode/allowing_apps_and_websites_to_link_to_your_content/preparing_your_app_to_be_the_default_browser_or_email_client)

**Comment by kelson42** [comment]
I have the same analysis. Kiwix is not a Web browser, therefore not concerned by the current "web browser app" approach.

This post is of interest as well https://dev.to/ben/will-ios-14-support-service-workers-in-wkwebview-5gn.

That said, the situation is evolving and it seems possible that WKWebView will support serive workers in the future.

For the moment, I believe this is important to ignore entries with the tag `_sw:yes`.

We will then re-evaluate the situation in a year.

**Comment by automactic** [comment]
> ignore entries with the tag _sw:yes

Just to be clear: you mean not to show these files in the library right?

**Comment by kelson42** [comment]
Yes this is what I mean. If open locally, the. The zim will display an error because it can itself detect that SW are not supported.

**Comment by automactic** [comment]
What happens if user try to manually open such a file?

**Comment by kelson42** [comment]
Not sure, but a message should be displayed that to be displayed this ZIM file needs service workers and this is not available.

**Comment by stephane-archer** [comment]
Do you think Mac Os has the same limitation as iOS? Do you have more alternatives than WKWebView?

**Comment by BenjaminJMueller** [comment]
Can we expect an update here in the following months? Would be highly appreciated if Zimit files would work on iOS and macOS soon.

**Comment by kelson42** [comment]
@rgaudin I have seen https://github.com/ionic-team/capacitor/issues/4122 which seems to confirm that SW are available for app-bounds-domain. That means that we could use such a bound-domain to allow SW:
* Either via a custom protocol handler (but SW has to work in that context)
* OR like we do on Kiwix for Android with our fake domain `kiwix.app`

**Comment by rgaudin** [comment]
Both needs to be tested as it's not guaranteed that it would be allowed
- We're not sure if SW would work with custom schemes. Security Context is HTTP specific
- We're not sure it it there's a mechanism similar to Andoid for intercepting requests `shouldInterceptRequest` and it it's as relaxed: we declare being on an HTTPS URL but we intercept/forge all requests (secure context here is not real)

**Comment by kelson42** [comment]
It seems that we will be able to create [ZIM files without SW with Zimit in version 2.0](https://github.com/openzim/zimit/issues/193). Therefore this feature request priority is significantly lowered.

**Comment by kelson42** [comment]
Closing as this won’t be necessary anymore once zimit2 will be released.
